File: typescript/shared/utils/timing.ts
/**
 * Timing Utility
 * High-precision timing for performance measurement
 */

export function startTiming(): number {
  return performance.now();
}

export function stopTiming(startTime: number): number {
  const elapsed = performance.now() - startTime;
  return Math.floor(elapsed * 100) / 100;
}

export function formatTiming(label: string, startTime: number): string {
  const elapsed = stopTiming(startTime);
  return `${label}: ${elapsed.toFixed(2)} ms`;
}

File: lua/shared/utils/timing.lua
---
--- Timing Utility
--- High-precision timing for performance measurement
---

local timing_utils = {}

---@return number start_time
function timing_utils.start()
  return vim.uv.hrtime()
end

---@param start_time number
---@return number elapsed_ms
function timing_utils.stop(start_time)
  local elapsed_ns = vim.uv.hrtime() - start_time
  return math.floor((elapsed_ns / 1e6) * 100) / 100
end

---@param label string
---@param start_time number
---@return string
function timing_utils.format(label, start_time)
  local elapsed = timing_utils.stop(start_time)
  return string.format("%s: %.2f ms", label, elapsed)
end

return timing_utils
